cmd/smoke_test.py

```diff
@@ -51,96 +45,47 @@ def signal_handler(sig, frame):
-    logging.debug("file_path: %s", file_path)
-
-    logging.info("Running %s...", (suite_name))
-
-    default_url = ""
-
-    if override_url is not None and len(override_url) > 0:
-        default_url = override_url
-    else:
-        default_url = configured_url
-
-    logging.debug("default_url: %s", default_url)
-    logging.debug("override_url: %s", override_url)
-    logging.debug("configured_url: %s", configured_url)
-
     signal.signal(signal.SIGINT, signal_handler)
     signal.signal(signal.SIGHUP, signal_handler)
     signal.signal(signal.SIGTERM, signal_handler)
 
-    #######################
-    ##  RUN TESTS
-    #######################
-    test_case = unittest.TestCase()
-
-    verificationErrors = []
-    for test in test_paths:
-        url = urljoin(default_url, test["path"])
-        test["url"] = url
-        testing_msg = "Testing " + json.dumps(test)
-        logging.info(testing_msg)
-
-        try:
 
-            req = requests.request(url=url, method=str(test["method"]).upper(), data=test["body"],
-                                   headers=test["headers"])  # Could return requests.exceptions.RequestException
-
-            test_case.assertEqual(req.status_code, test["expected_status_code"],
-                                  "unexpected status code.")  # could return an AssertionError
-        except Exception as e:
-            verificationErrors.append(str(e))
-            fail_msg = "FAIL: " + str(e)
-            logging.error((fail_msg))
-            #logging.error(traceback.format_exc())
-
-            if exit_on_error:
-                logging.error("aborting the rest of the tests")
-                break
 
     #######################
-    ##  LOG RESULTS
+    ##  RUN TESTS
     #######################
-    final_string = ""
-    final_line = ""
-    main_error = "MAIN_ERRORS = " + str(len(verificationErrors))
-    logging.info(main_error)
-    if len(verificationErrors) > 0:
-        final_string = "FAIL: " + suite_name
-        final_line = "failed!"
-        fail_suite = "FAIL: " + suite_name + " ran with failures"
-        logging.error(final_string)
-        logging.error(final_line)
-        logging.error(fail_suite)
-    else:
-        final_string = "PASS: " + suite_name
-        final_line = "passed!"
-        final_suite = "PASS: " + suite_name + " passed!"
-        logging.info(final_string)
-        logging.info(final_line)
-        logging.info(final_suite)
-
-    # throw a non-zero exit code if there were errors!
-    exit(len(verificationErrors) > 0)
+    try:
+        cmd = ["pytest", "-vvvv", test_dir, "--smoke-json", smoke_json_file]
+        if override_url is not None and len(override_url) != 0:
+            cmd += ["--smoke-url", override_url]
+        if allure_dir is not None:
+            cmd += ["--alluredir", allure_dir]
+        if exit_on_first_failure:
+            cmd += ["-x"]
+        pytest_run = subprocess.run(cmd, check=True)
+    except Exception as e:
```

Comment:
The exception could be logged

Reply:
I think I'm not going to print the expectation, as it causes the output to look really bad when the tests failed:
```
24090-smoke_test-1  | 2023-01-13 15:49:11,212 FAIL
24090-smoke_test-1  | --- Logging error ---
24090-smoke_test-1  | Traceback (most recent call last):
24090-smoke_test-1  |   File "/src/app/smoke_test.py", line 86, in <module>
24090-smoke_test-1  |     pytest_run = subprocess.run(cmd, check=True)
24090-smoke_test-1  |   File "/usr/lib/python3.9/subprocess.py", line 528, in run
24090-smoke_test-1  |     raise CalledProcessError(retcode, process.args,
24090-smoke_test-1  | subprocess.CalledProcessError: Command '['pytest', '-vvvv', '/src/app/smoke_pytest', '--smoke-json', '/src/libs/example_smoke.json']' returned non-zero exit status 1.
24090-smoke_test-1  | 
24090-smoke_test-1  | During handling of the above exception, another exception occurred:
24090-smoke_test-1  | 
24090-smoke_test-1  | Traceback (most recent call last):
24090-smoke_test-1  |   File "/usr/lib/python3.9/logging/__init__.py", line 1083, in emit
24090-smoke_test-1  |     msg = self.format(record)
24090-smoke_test-1  |   File "/usr/lib/python3.9/logging/__init__.py", line 927, in format
24090-smoke_test-1  |     return fmt.format(record)
24090-smoke_test-1  |   File "/usr/lib/python3.9/logging/__init__.py", line 663, in format
24090-smoke_test-1  |     record.message = record.getMessage()
24090-smoke_test-1  |   File "/usr/lib/python3.9/logging/__init__.py", line 367, in getMessage
24090-smoke_test-1  |     msg = msg % self.args
24090-smoke_test-1  | TypeError: not all arguments converted during string formatting
24090-smoke_test-1  | Call stack:
24090-smoke_test-1  |   File "/src/app/smoke_test.py", line 90, in <module>
24090-smoke_test-1  |     logging.error("Exception:", e)
24090-smoke_test-1  | Message: 'Exception:'
24090-smoke_test-1  | Arguments: (CalledProcessError(1, ['pytest', '-vvvv', '/src/app/smoke_pytest', '--smoke-json', '/src/libs/example_smoke.json']),)
```

The exception will always be thrown when the smoke tests fail, and might cause confusion on were the actually problem is.

Reply:
That sounds good. Thanks for trying it.